### DISCUSSION TITLE
Can't set specific arguments to a matrix

### DISCUSSION THREAD

**Discussion by JosephBoom02** [typst/typst#3045] [Q&A] [answered]
So, I want to set the delimiter of a matrix to be "[" and the gap 1em. If I input only the delimiter argument the matrix is well formed:
```
mat(delim: "[",
        - lambda , 1;  
        -  frac(k,M) , - frac(h,M) -  lambda
    )
```
but, if I try to insert ```gap: 1em```,  typst return the error: "unknown variable: em".
So. I tried to write this
```
#math.equation(numbering: none, block: true)[
$    
    A &= 
    #math.mat(delim: "[", gap: 1em,
        $ 
        0 , 1;  
        -  frac(k,M) , - frac(h,M)
        $
    )
$]
```
but the output is not a matrix:
![Screenshot from 2023-12-22 14-50-47](https://github.com/typst/typst/assets/30532436/e425d104-4f6c-419d-886a-3b2a5ea668d7)
What should I do to achieve the result I want?

**Comment by HollowNumber** [comment]
To set a gap for the matrices you can use a `set` rule:

```typ
#set math.mat(gap: 1em)

$mat(delim: "[",
        - lambda, 1;  
        - frac(k,M), - frac(h,M) - lambda)
$    

```

Furthermore, if you want to make the delimeter be square brackets for all future matrices you can also put that in the `set` rule

```typ
#set math.mat(gap: 1em, delim: "[")

$
A &= mat(
0, 1;
- frac(k,M), - frac(h,M);
)
$
```

Hope this helps!

**Reply by JosephBoom02** [reply]
OK, but if I want to set those options only for one matrix and not for the other?
N.B. I tried to use the show rule with the label, so only matrices with that label use the options I specify:
```
#show <big_matrices>: set math.mat(delim: "[", gap: 2em)
```
but I also set, with ```i-figured``` package, that only equations with the label are numbered. So, if I use that show rule the matrix I want to have more gaps is numbered, and I don't want this.

**Comment by xkevio** [comment, marked as answer]
You have to prefix the `1em` with `#` to temporarily return to code mode as it will otherwise interpret `em` as a variable name.
```typ
$    
    A = mat(delim: "[", gap: #1em, 
      0, 1;  
      -frac(k,M), -frac(h,M)
    )
$
```

**Reply by JosephBoom02** [reply]
Thanks
